src/openassetio-core/include/openassetio/errors.h

```diff
@@ -29,8 +29,11 @@
 /// Unknown error when processing a batch element.
 #define OPENASSETIO_BatchErrorCode_kUnknown (OPENASSETIO_BatchErrorCode_BEGIN + 0)
 
-/// Entity resolution failure.
-#define OPENASSETIO_BatchErrorCode_kEntityResolutionError (OPENASSETIO_BatchErrorCode_BEGIN + 1)
-
 /// Failure due to malformed entity reference.
-#define OPENASSETIO_BatchErrorCode_kInvalidEntityReference (OPENASSETIO_BatchErrorCode_BEGIN + 2)
+#define OPENASSETIO_BatchErrorCode_kInvalidEntityReference (OPENASSETIO_BatchErrorCode_BEGIN + 1)
+
+/// Failure due to an API method being called with an invalid @ref Context access.
+#define OPENASSETIO_BatchErrorCode_kEntityAccessError (OPENASSETIO_BatchErrorCode_BEGIN + 2)
+
+/// Entity resolution failure.
+#define OPENASSETIO_BatchErrorCode_kEntityResolutionError (OPENASSETIO_BatchErrorCode_BEGIN + 3)
```

Comment:
[question] I assume this is something todo with the C++ and Python interop, but why not have these as an enum instead of #defines?

Reply:
A good q! We also have a [`C` API](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/src/openassetio-core-c/include/openassetio/c/errors.h), which precludes `enum class`, and so we wanted a single source of truth for the actual values, to avoid inadvertent drift.  We haven't added the equivalent of the `BatchElementError` to that API yet though.